src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -425,6 +444,45 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
             }
         }
 
+        if ($flattenNestedAttributes) {
```

Comment:
This check can be removed with my proposal (this reduces the cyclomatic complexity).